docs/tutorial.md

```diff
@@ -0,0 +1,227 @@
+Callback usually takes parameter called `err` as the first and the second (if specified) is a value that is passed to the callback,
+if `err` is not `null` then error must be happened in the main function.
+
+Examples :
+
+#### Craft Stick Bot
+
+Here we're crafting oak logs into sticks.
+Craft oak logs into oak wood planks, wait for crafting finish. Then craft the oak wood planks into sticks.
+
+Incorect aproach ❌ :
+
+```js
+const plankRecipe = bot.recipesFor(5); // 5 ID for Oak Wood Planks
+bot.craft(plankRecipe, 1); // ❌
+const stickRecipe = bot.recipesFor(280); // 280 ID for Stick
+bot.craft(plankRecipe, 1); // ❌
+// You don't wanna start crafting sticks, before crafting logs into planks finished.
+```
+
+Correct approach with callbacks ✔️ :
+
+```js
+const plankRecipe = bot.recipesFor(5)[0]; // 5 ID for Oak Wood Planks
+
+bot.craft(plankRecipe, 1, null, function (err) {
+  // if bot.craft(plankRecipe, ...) then the code bellow get executed ✔️
+  if (err) return bot.chat(err.message);
+  // if error happened when crafting `plankRecipe` then send to chat the `err.message`
+
+  const stickRecipe = bot.recipesFor(280); // 280 ID for Stick
+  bot.craft(stickRecipe, 1, null, function (err) {
+    if (err) return bot.chat(err.message);
+    // if error happened when crafting `stickRecipe` then send to chat the `err.message`
+
+    bot.chat("Crafting Sticks finished");
+  });
+});
+```
+
+More on [`bot.craft()`](http://mineflayer.prismarine.js.org/#/api?id=botcraftrecipe-count-craftingtable-callback) method.
+
+#### Consuming Bot
+
+In this example we create bot that eat / consume held item if hungry.
+Using
+[`health`](http://mineflayer.prismarine.js.org/#/api?id=health) event of the bot to listen on any changes in health or food points.
+[`bot.consume()`](http://mineflayer.prismarine.js.org/#/api?id=botconsumecallback) method to eat.
+[`bot.food`](http://mineflayer.prismarine.js.org/#/api?id=botfood) property to check food points.
+`bot.heldItem` property to get the name of the item held by the bot. `bot.heldItem` is an instance of [`Item`](https://github.com/PrismarineJS/prismarine-item/blob/master/README.md) or `null` if no item held.
+
+```js
+const minFoodPoints = 18;
+
+bot.on("health", function () {
+  // listen on any changes in health or food points.
+  if (bot.food < minFoodPoints) {
+    // if the food points lower than 18, then
+    const heldItemName = bot.heldItem.name;
+
+    function onFinishEating(err) {
+      if (err) return bot.chat(err.message);
+      bot.chat(`I'm finish eating ${heldItemName}`);
+    }
+
+    bot.chat(`Eating ${heldItemName}`);
+    bot.consume(onFinishEating);
+    // `bot.consume()` method takes `onFinishEating` as callback, and execute `onFinishEating` when its finish or stumble an error.
+  }
+});
+```
+
+#### Fisherman Bot
+
+Can be found in [fishing.js](https://github.com/PrismarineJS/mineflayer/blob/master/examples/fisherman.js#L52).
```

Comment:
It's not a tutorial not a guide. It's just a link to an example

Reply:
Yeah. ill remove that.